Using multi build on local source dir is not working

Using multi with --ffmpeg-source-dir
```
./cross_compile_ffmpeg.sh --ffmpeg-source-dir=$GITHUB_WORKSPACE/ffmpeg-source --gcc-cpu-count=2 --disable-nonfree=n --sandbox-ok=y --compiler-flavors=multi
```
Returns with
```
2020-07-14T20:43:15.0520578Z WARNING: We thought we found pkg-config '/home/runner/work/ffmpeg-autobuild/ffmpeg-autobuild/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32-pkg-config' but now it's not there. How odd!
2020-07-14T20:43:15.0520834Z Found Pkg-config: NO
2020-07-14T20:43:15.0520941Z Found CMake: NO
2020-07-14T20:43:15.0521181Z Run-time dependency libffi found: NO (tried pkgconfig and cmake)
2020-07-14T20:43:15.0521297Z Library ffi found: YES
2020-07-14T20:43:15.0521397Z Has header "ffi.h" : YES 
2020-07-14T20:43:15.0521633Z Run-time dependency zlib found: YES 1.2.11
2020-07-14T20:43:15.0521744Z Checking for function "ngettext" : NO 
2020-07-14T20:43:15.0521848Z Library intl found: YES
2020-07-14T20:43:15.0522110Z Checking for function "bind_textdomain_codeset" with dependency -lintl: NO 
2020-07-14T20:43:15.0522233Z Library ws2_32 found: YES
2020-07-14T20:43:15.0522326Z Checking for function "strlcpy" : NO 
2020-07-14T20:43:15.0522434Z Program python3 found: YES (/usr/bin/python3)
2020-07-14T20:43:15.0522543Z Program bash found: YES
2020-07-14T20:43:15.0522643Z Program sh found: YES
2020-07-14T20:43:15.0522742Z Program env found: YES
2020-07-14T20:43:15.0522912Z Configuring glibconfig.h using configuration
2020-07-14T20:43:15.0523022Z Fetching value of define "arm" :  
2020-07-14T20:43:15.0523115Z Fetching value of define "__arm" :  
2020-07-14T20:43:15.0523222Z Fetching value of define "__arm__" :  
2020-07-14T20:43:15.0523506Z Compiler for C supports arguments -Wno-format-nonliteral -Wformat-nonliteral: YES 
2020-07-14T20:43:15.0523797Z Compiler for C supports arguments -Wno-duplicated-branches -Wduplicated-branches: YES 
2020-07-14T20:43:15.0523881Z 
2020-07-14T20:43:15.0523997Z glib/gnulib/meson.build:277:2: ERROR: Problem encountered: frexp() is missing or broken beyond repair, and we have nothing to replace it with
2020-07-14T20:43:15.0524090Z 
2020-07-14T20:43:15.0524396Z A full log can be found at /home/runner/work/ffmpeg-autobuild/win64/glib-2.64.3/build/meson-logs/meson-log.txt
2020-07-14T20:43:15.1377176Z ##[error]Process completed with exit code 1.
```
[Full Log.txt.gz](https://github.com/rdp/ffmpeg-windows-build-helpers/files/4942168/Full.Log.txt.gz)
Running this script on GitHub Actions with Ubuntu 18.04

--- a/cross_compile_ffmpeg.sh
+++ b/cross_compile_ffmpeg.sh
@@ -2436,7 +2436,12 @@ build_ffmpeg() {
       echo "You will find redistributable archive .7z file in $cur_dir/redist"
     fi
     echo `date`
-  cd ..
+    
+  if [[ -z $ffmpeg_source_dir ]]; then
+    cd ..
+  else
+    cd "$work_dir"
+  fi
 }
 
 build_lsw() {
@@ -2789,8 +2794,9 @@ if [[ $compiler_flavors == "native" ]]; then
   #  bs2b doesn't use pkg-config, sndfile needed Carbon :|
   export CPATH=$cur_dir/cross_compilers/native/include:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Carbon.framework/Versions/A/Headers # C_INCLUDE_PATH
   export LIBRARY_PATH=$cur_dir/cross_compilers/native/lib
-  mkdir -p native
-  cd native
+  work_dir="$(realpath $cur_dir/native)"
+  mkdir -p "$work_dir"
+  cd "$work_dir"
     build_ffmpeg_dependencies
     build_ffmpeg
   cd ..
@@ -2809,8 +2815,9 @@ if [[ $compiler_flavors == "multi" || $compiler_flavors == "win32" ]]; then
   bits_target=32
   cross_prefix="$mingw_bin_path/i686-w64-mingw32-"
   make_prefix_options="CC=${cross_prefix}gcc AR=${cross_prefix}ar PREFIX=$mingw_w64_x86_64_prefix RANLIB=${cross_prefix}ranlib LD=${cross_prefix}ld STRIP=${cross_prefix}strip CXX=${cross_prefix}g++"
-  mkdir -p win32
-  cd win32
+  work_dir="$(realpath $cur_dir/win32)"
+  mkdir -p "$work_dir"
+  cd "$work_dir"
     build_ffmpeg_dependencies
     build_apps
   cd ..
@@ -2829,8 +2836,9 @@ if [[ $compiler_flavors == "multi" || $compiler_flavors == "win64" ]]; then
   bits_target=64
   cross_prefix="$mingw_bin_path/x86_64-w64-mingw32-"
   make_prefix_options="CC=${cross_prefix}gcc AR=${cross_prefix}ar PREFIX=$mingw_w64_x86_64_prefix RANLIB=${cross_prefix}ranlib LD=${cross_prefix}ld STRIP=${cross_prefix}strip CXX=${cross_prefix}g++"
-  mkdir -p win64
-  cd win64
+  work_dir="$(realpath $cur_dir/win64)"
+  mkdir -p "$work_dir"
+  cd "$work_dir"
     build_ffmpeg_dependencies
     build_apps
   cd ..

